### PR TITLE
MNT: Fix component names

### PR DESCRIPTION
**Pull request by klauer**
Missed this one in the plugins PR. Notice any others?

### CODE REVIEW

File: ophyd/areadetector/plugins.py [hunk 1]
```diff
@@ -2149,7 +2149,7 @@ class CodecPlugin(Device, version_type='ADCore'):
 
 
 class CodecPlugin_V34(PluginBase_V34, CodecPlugin, version=(3, 4), version_of=CodecPlugin):
-    blosc_cl_evel = Cpt(SignalWithRBV, "BloscCLevel")
+    blosc_compression_level = Cpt(SignalWithRBV, "BloscCLevel")
```
Comment (mrakitin) [inline ophyd/areadetector/plugins.py]: 👍

**Comment by klauer** [comment]
@mrakitin - nope, just taking a closer look it popped out as incorrect. A few more eyes on the component names would be good, prior to the next release.

**Comment by mrakitin** [comment]
@klauer, I see, thanks. I guess we cannot easily perform smoke tests for them as we don't have real devices.

Will look through the components as well. Is it only in plugins?

**Comment by klauer** [comment]
@mrakitin - note that the PVs themselves are correct, this is just the attribute names. It's not something you can really test...

That said, assuming my coverage tests and the associated PV lists (https://github.com/klauer/adcoverage/blob/master/coverage.py) are correct, there shouldn't be any issues with PV names for all of the plugins.

**Comment by prjemian** [comment]
I agree the naming of things is not testable.  What revealed the problem to *me* was the fragment `bl_evel` in the name.  This did not fit the usual practice of naming involving short hand, abbreviations, acronyms, camelCasing, ...   Just [*kinda funny looking*](https://www.imdb.com/title/tt0116282/?ref_=fn_al_tt_2).  Good catch, @klauer!

**Comment by mrakitin** [comment]
This does not look right: https://github.com/NSLS-II/ophyd/blob/579b2ba9a99f935bd976768e7450f9f933fe8a8a/ophyd/areadetector/plugins.py#L1493-L1494

I believe it should be `read_background_tiff_seq` and `read_flat_field_tiff_seq`.

**Comment by klauer** [comment]
Fixed, @mrakitin, thanks! And thanks for taking a look, @danielballan.

**Comment by prjemian** [comment]
Looks good now.  Nothing stands out.

**Comment by tacaswell** [comment]
grep "Cpt.*'.*[A-Z][A-Z]" ophyd/areadetector/plugins.py 


```
amilyn@15:13  ➤  grep "Cpt.*'.*[A-Z][A-Z]" ophyd/areadetector/plugins.py 
    array_rate = Cpt(EpicsSignalRO, 'ArrayRate_RBV')
    asyn_io = Cpt(EpicsSignal, 'AsynIO')
    nd_attributes_file = Cpt(EpicsSignal, 'NDAttributesFile', string=True)
    port_name = Cpt(EpicsSignalRO, 'PortName_RBV', string=True, kind='config')
    width = Cpt(EpicsSignalRO, 'ArraySize0_RBV')
    height = Cpt(EpicsSignalRO, 'ArraySize1_RBV')
    depth = Cpt(EpicsSignalRO, 'ArraySize2_RBV')
    bayer_pattern = Cpt(EpicsSignalRO, 'BayerPattern_RBV')
    color_mode = Cpt(EpicsSignalRO, 'ColorMode_RBV')
    data_type = Cpt(EpicsSignalRO, 'DataType_RBV', string=True)
    dim0_sa = Cpt(EpicsSignal, 'Dim0SA')
    dim1_sa = Cpt(EpicsSignal, 'Dim1SA')
    dim2_sa = Cpt(EpicsSignal, 'Dim2SA')
    dimensions = Cpt(EpicsSignalRO, 'Dimensions_RBV')
    nd_array_address = Cpt(SignalWithRBV, 'NDArrayAddress')
    nd_array_port = Cpt(SignalWithRBV, 'NDArrayPort', kind='config')
    ndimensions = Cpt(EpicsSignalRO, 'NDimensions_RBV')
    plugin_type = Cpt(EpicsSignalRO, 'PluginType_RBV', lazy=False, kind='config')
    queue_use_high = Cpt(EpicsSignal, 'QueueUseHIGH')
    queue_use_hihi = Cpt(EpicsSignal, 'QueueUseHIHI')
    time_stamp = Cpt(EpicsSignalRO, 'TimeStamp_RBV')
    unique_id = Cpt(EpicsSignalRO, 'UniqueId_RBV')
    hist_entropy = Cpt(EpicsSignalRO, 'HistEntropy_RBV', kind='config')
    histogram = Cpt(EpicsSignalRO, 'Histogram_RBV')
    max_value = Cpt(EpicsSignalRO, 'MaxValue_RBV')
    mean_value = Cpt(EpicsSignalRO, 'MeanValue_RBV')
    min_value = Cpt(EpicsSignalRO, 'MinValue_RBV')
    net = Cpt(EpicsSignalRO, 'Net_RBV')
    set_xhopr = Cpt(EpicsSignal, 'SetXHOPR')
    set_yhopr = Cpt(EpicsSignal, 'SetYHOPR')
    sigma_xy = Cpt(EpicsSignalRO, 'SigmaXY_RBV')
    sigma_x = Cpt(EpicsSignalRO, 'SigmaX_RBV')
    sigma_y = Cpt(EpicsSignalRO, 'SigmaY_RBV')
    sigma = Cpt(EpicsSignalRO, 'Sigma_RBV')
    ts_acquiring = Cpt(EpicsSignal, 'TSAcquiring')
    ts_control = Cpt(EpicsSignal, 'TSControl', string=True, kind='config')
    ts_current_point = Cpt(EpicsSignal, 'TSCurrentPoint')
    ts_max_value = Cpt(EpicsSignal, 'TSMaxValue')
    ts_mean_value = Cpt(EpicsSignal, 'TSMeanValue')
    ts_min_value = Cpt(EpicsSignal, 'TSMinValue')
    ts_net = Cpt(EpicsSignal, 'TSNet')
    ts_num_points = Cpt(EpicsSignal, 'TSNumPoints', kind='config')
    ts_read = Cpt(EpicsSignal, 'TSRead')
    ts_sigma = Cpt(EpicsSignal, 'TSSigma')
    ts_sigma_x = Cpt(EpicsSignal, 'TSSigmaX')
    ts_sigma_xy = Cpt(EpicsSignal, 'TSSigmaXY')
    ts_sigma_y = Cpt(EpicsSignal, 'TSSigmaY')
    ts_total = Cpt(EpicsSignal, 'TSTotal')
    total = Cpt(EpicsSignalRO, 'Total_RBV')
    foffset = Cpt(SignalWithRBV, 'FOffset', kind='config')
    fscale = Cpt(SignalWithRBV, 'FScale', kind='config')
    num_filtered = Cpt(EpicsSignalRO, 'NumFiltered_RBV', kind='config')
    o_offset = Cpt(SignalWithRBV, 'OOffset', kind='config')
    o_scale = Cpt(SignalWithRBV, 'OScale', kind='config')
    roffset = Cpt(SignalWithRBV, 'ROffset', kind='config')
    valid_background = Cpt(EpicsSignalRO, 'ValidBackground_RBV', string=True, kind='config')
    valid_flat_field = Cpt(EpicsSignalRO, 'ValidFlatField_RBV', string=True, kind='config')
    position_xlink = Cpt(EpicsSignal, 'PositionXLink')
    position_ylink = Cpt(EpicsSignal, 'PositionYLink')
    set_xhopr = Cpt(EpicsSignal, 'SetXHOPR')
    set_yhopr = Cpt(EpicsSignal, 'SetYHOPR')
    size_xlink = Cpt(EpicsSignal, 'SizeXLink')
    size_ylink = Cpt(EpicsSignal, 'SizeYLink')
    name_ = Cpt(SignalWithRBV, 'Name', doc='ROI name', kind='config')
    set_xhopr = Cpt(EpicsSignal, 'SetXHOPR')
    set_yhopr = Cpt(EpicsSignal, 'SetYHOPR')
    file_path_exists = Cpt(EpicsSignalRO, 'FilePathExists_RBV', kind='config')
    full_file_name = Cpt(EpicsSignalRO, 'FullFileName_RBV', string=True, kind='config')
    num_captured = Cpt(EpicsSignalRO, 'NumCaptured_RBV')
    jpeg_quality = Cpt(SignalWithRBV, 'JPEGQuality', kind='config')
    io_speed = Cpt(EpicsSignal, 'IOSpeed', kind='config')
    szip_num_pixels = Cpt(SignalWithRBV, 'SZipNumPixels', kind='config')
    zlevel = Cpt(SignalWithRBV, 'ZLevel', kind='config')
    array_data = Cpt(EpicsSignalRO, 'ArrayData_RBV')
    value_sum = Cpt(EpicsSignalRO, 'ValueSum_RBV')
    value = Cpt(EpicsSignalRO, 'Value_RBV')
    ts_acquiring = Cpt(EpicsSignal, 'TSAcquiring', string=True, doc="0='Done' 1='Acquiring'")
    ts_control = Cpt(EpicsSignal, 'TSControl', string=True, doc="0=Erase/Start 1=Start 2=Stop 3=Read")
    ts_current_point = Cpt(EpicsSignal, 'TSCurrentPoint')
    ts_num_points = Cpt(EpicsSignal, 'TSNumPoints')
    ts_read = Cpt(EpicsSignal, 'TSRead')
    ts_control = Cpt(EpicsSignal, 'TSControl', string=True, doc="0=Erase/Start 1=Start 2=Stop 3=Read")
    ts_current_point = Cpt(EpicsSignal, 'TSCurrentPoint')
    ts_num_points = Cpt(EpicsSignal, 'TSNumPoints')
    ts_read = Cpt(EpicsSignal, 'TSRead')

```

**Comment by tacaswell** [comment]
looks like there are a couple more that are not quite right:

    set_xhopr = Cpt(EpicsSignal, 'SetXHOPR')
    set_yhopr = Cpt(EpicsSignal, 'SetYHOPR')
    size_xlink = Cpt(EpicsSignal, 'SizeXLink')
    size_ylink = Cpt(EpicsSignal, 'SizeYLink')
    szip_num_pixels = Cpt(SignalWithRBV, 'SZipNumPixels', kind='config')

**Comment by klauer** [comment]
@tacaswell I think those look okayish. `set_x_hopr` would probably be preferable, but since it's in R1-9-1 I think we should leave it for back-compat. Same with `*link`.

For the szip one, it's actually correct - `szip` is the library.

**Comment by prjemian** [comment]
* `set_x_hopr` *v.* `set_xhopr` *v.* `setxhopr` : it's a toss-up
* `size_xlink` : *op cit.*
* `szip` is a well-known software package: `szip_num_pixels` seems OK

not far off enough to change (IMO)

**Comment by mrakitin** [comment]
Here is my trivial comparison of the component names with the PV names based on conversion to `underscore` performed by [inflection](https://inflection.readthedocs.io/en/latest/#inflection.underscore): https://gist.github.com/mrakitin/13c810ef5dc8da722f80c4c200625e2d. It does not fully cover all the cases since there are new lines (`\n`), DDC's, etc. Here is a summary:
```
'enable' : 'enable_callbacks'
'foffset' : 'f_offset'
'fscale' : 'f_scale'
'roffset' : 'r_offset'
'overlay_portname' : 'name'
'position_xlink' : 'position_x_link'
'position_ylink' : 'position_y_link'
'size_xlink' : 'size_x_link'
'size_ylink' : 'size_y_link'
'szip_num_pixels' : 's_zip_num_pixels'
'zlevel' : 'z_level'
'type_' : 'type'
'attribute_name' : 'attr_name'
```
(I skipped `overlay_#`, `width`, `height`, `depth`, `shaped_image`)

As for the `set_xhopr`, inflection converts `SetXHOPR` to `set_xhopr`, so I assume it's OK as is.

**Comment by prjemian** [comment]
mostly OK but `'szip_num_pixels' : 's_zip_num_pixels'` is a definite *no*, the well-known software is *szip*, the proposed revision suggests otherwise

**Comment by klauer** [comment]
Thanks, @mrakitin. Looks like we're pretty good, then.

All of the ones I'd consider changing are in the base plugin class (i.e., R-1-9-1) and not what was recently added for newer plugin versions R-2-0+, so I'm going to leave those as-is.

The rest are intentional renames to avoid clashing with ophyd/Python builtins.

**Comment by mrakitin** [comment]
Thanks for considering, @klauer. Yes, makes sense to keep the same names for compatibility.

**Pull request merged**